### PR TITLE
[CIGolang]timeout for disruptive

### PR DESCRIPTION
**Pull request by kuiwang02**
/cc @Xia-Zhao-rh

### CODE REVIEW

**Comment by kuiwang02** [comment]
@smg247 
Hi, the timeout of openshift-tests-private repo prow job is 8h by default currently. Per openshift QE testing requirement, we need to add test for destructive and long duration test, and then the 8h is not enough, especially for long duration test late (please refer to [the log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-ipi-destructive-p3/1565127470674350080) see the destructive test failing due to timeout.). 

Here is my solution: 
1, change 8h to 23h in [config file](https://github.com/openshift/release/blob/master/core-services/prow/02_config/_config.yaml) for openshift-tests-private repo by default. (if we keep 8h by default, we meet error `invalid configuration: tests[0]: job timeout is limited to 8h0m0s` when adding `timeout: 23h0m0s` for destructive and long duration test job)
2, add `timeout: 8h0m0s` for the test job except for destructive and long duration test prow job to keep origin timeout for these test jobs
3,  let destructive and long duration test job to take default value of repo prow job (23h)

I make trial with PR as example to approve my solution. but I find the destructive test still stop by 8h(please refer to [the log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/31934/rehearse-31934-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-ipi-destructive-p3/1565658409737392128) which does not take 23h as timeout)

So, maybe please help me (if you are not right person, please try to redirect me, thanks).
1, is my solution acceptable? or other better solution?
2, if it is acceptable, why the new default value does not take effective? 

Thanks in advance!

cc @chengzhang1016

**Comment by kuiwang02** [comment]
> Details

@smg247 
for 2), it seems works per [the new destrutive tet log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/31934/rehearse-31934-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-ipi-destructive-p3/1566223852344709120). I will try it more and update the info.
for 1), please still help check it. thanks

**Comment by kuiwang02** [comment]
> @smg247 Hi, the timeout of openshift-tests-private repo prow job is 8h by default currently. Per openshift QE testing requirement, we need to add test for destructive and long duration test, and then the 8h is not enough, especially for long duration test late (please refer to [the log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-ipi-destructive-p3/1565127470674350080) see the destructive test failing due to timeout.).
> 
> Here is my solution: 1, change 8h to 23h in [config file](https://github.com/openshift/release/blob/master/core-services/prow/02_config/_config.yaml) for openshift-tests-private repo by default. (if we keep 8h by default, we meet error `invalid configuration: tests[0]: job timeout is limited to 8h0m0s` when adding `timeout: 23h0m0s` for destructive and long duration test job) 2, add `timeout: 8h0m0s` for the test job except for destructive and long duration test prow job to keep origin timeout for these test jobs 3, let destructive and long duration test job to take default value of repo prow job (23h)
> 
> I make trial with PR as example to approve my solution. but I find the destructive test still stop by 8h(please refer to [the log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/31934/rehearse-31934-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-ipi-destructive-p3/1565658409737392128) which does not take 23h as timeout)
> 
> So, maybe please help me (if you are not right person, please try to redirect me, thanks). 1, is my solution acceptable? or other better solution? 2, if it is acceptable, why the new default value does not take effective?
> 
> Thanks in advance!
> 
> cc @chengzhang1016

@smg247 
for 2), it seems 23h works per [the new destrutive tet log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/31934/rehearse-31934-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-aws-ipi-destructive-p3/1566223852344709120). I will try it more and update the info.
for 1), please still help check it. thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 1]
```diff
@@ -65,6 +65,7 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-aws-ipi
+  timeout: 8h0m0s
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:68]: We should remove the limit here.

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:68]: we change default value to 23h, so add 8h for non-desctrucitve test and long duration test.
please refer to https://github.com/openshift/release/pull/31934#issuecomment-1236007314 and https://github.com/openshift/release/pull/31934#issuecomment-1236333913

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:68]: `openshift-e2e-test-qe` runs multiple tests, ginkgo (in openshift-tests-private) is only part of it. 
Set ginkgo to timeout 23 hour, while set the total to timeout 8 hour, it does not looks a great plan IMO.

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:68]: it does not change ginkgo step to 23h. it is for test level. I change to 23h in the code https://github.com/openshift/release/pull/31934/files#diff-3fe4522c3fa00ecd918a986bc31b4cde479c02bcace2b0497bd136760edbd71f

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:68]: we use one step to cover destructive, longduration, and other cases. although the max timetout is 23h for the step, it does not run so long time for non-destructive, and non-longduration, so it is OK to add 8h for non-destructive, and non-longduration test.

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:68]: so, you suggest we make different step for different case, and set different timeout for step, so we do not need specific timeout for test?

**Comment by kuiwang02** [comment]
Per https://github.com/openshift/release/pull/31934#discussion_r962524519, I make different golang step for diffrent kind of test case: step for disruptive, step for longduration, and step for other cases.
and then set different timeout for different step.
So, we only need to increase the timeout for test of openshift-tests-private repo to 23h and depends on step timeout to limit duration. and then
1, we do not need to add timeout: 8h0m0s for each non-disruptive and non-longduration test, and while still limit these test not up to 23h.
2, for non-disruptive and non-longduration test, expect it could be up to 23h (default value of test of repo)

**Comment by kuiwang02** [comment]
@smg247 @dptp 
Hi, could you please review it and add lgtm? thanks.

current solution:
1, we define the different step for destructive, longduration and non-destructive-longduration case. and set different limitation timeout at step.
2, we have to change test level limitation timeout to 23h for openshift-tests-private repo so that the destructive test and longduration test is not stopped.

cc @chengzhang1016

**Pull request merged**